sdk/keyvault/azure-security-keyvault-certificates/src/certificate_client_operations.cpp

```diff
@@ -59,22 +66,22 @@ std::unique_ptr<Azure::Core::Http::RawResponse> CreateCertificateOperation::Poll
 
   if (m_status == Azure::Core::OperationStatus::Succeeded)
```

Comment:
I think there's a bug about how we are setting the m_status:

![image](https://user-images.githubusercontent.com/24213737/140173629-ee1972a4-8add-499e-9e5f-d20f4fca9b23.png)


We are using the `HttpStatusCode` from the operation response to se the m_status.  We should be parsing the OperationProperties to know about the operation current status and use it to set m_status.

Right now, as soon as we get a valid Operation response, we are setting m_status == Succeded, but the operation might have come with a payload that contains status: 'pending'.

Reply:
we discussed this in a call